bgls/simulator.py

```diff
@@ -201,3 +210,23 @@ def _sample_bitstrings(
                     )
                 ]
         return keys_to_bitstrings
+
+    def _needs_trajectories(self, circuit: "cirq.AbstractCircuit") -> bool:
+        """Determines if repeated samples can be drawn for a single
+        simulation. For near-clifford, noisy, or non-unitary circuits this
+        is not possible. See qsim/qsimcirq/qsim_simulator.py"""
```

Comment:
"See qsim/qsimcirq/qsim_simulator.py" is enough, but "Taken from https://github.com/quantumlib/qsim/blob/235ae2fc039fb4a98beb4a6114d10c7f8d2070f7/qsimcirq/qsim_simulator.py#L29" is also an option. As in qsim this is really a static method and could be outside the class.

Reply:
Got it yeah I just added apply_gate as a parameter and was able to move it outside the class